src/transformers/trainer.py

```diff
@@ -644,6 +656,13 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
             else:
                 self.optimizer = optimizer_cls(optimizer_grouped_parameters, **optimizer_kwargs)
 
+    def create_scheduler(self, num_training_steps: int):
+        """
+        Setup the scheduler. The optmizer of the trainer must have been set up.
```

Comment:
```suggestion
        Setup the scheduler. The optimizer of the trainer must have been set up before this method is called.
```

Add also the mention of subclassing and overriding if possible.